hearth/spec/hearth/http/middleware/request_compression_spec.rb

```diff
@@ -81,24 +81,59 @@ def expect_uncompressed_body(request, body)
             end
           end
 
-          context 'no supported encodings' do
-            let(:encodings) { ['custom'] }
+          context 'encodings' do
+            context 'none supported by client' do
+              let(:encodings) { ['custom'] }
 
-            it 'skips compression' do
-              expect(app).to receive(:call).with(input, context)
-              resp = subject.call(input, context)
-              expect_uncompressed_body(request, body)
-              expect(resp).to be output
+              it 'skips compression' do
+                expect(app).to receive(:call).with(input, context)
+                resp = subject.call(input, context)
+                expect_uncompressed_body(request, body)
+                expect(resp).to be output
+              end
             end
-          end
 
-          context 'multiple encodings' do
-            let(:encodings) { %w[custom gzip] }
+            context 'multiple encodings' do
+              let(:encodings) { %w[custom gzip] }
+              let(:body) { StringIO.new('a' * 16_385) }
+
+              it 'processes the first supported encoding found' do
+                expect(app).to receive(:call).with(input, context)
+                resp = subject.call(input, context)
+                expect(request.headers['Content-Encoding']).to eq('gzip')
+                expect(resp).to be output
+              end
+            end
+
+            context 'unsupported encoding is given' do
+              it 'raises a standard error in compression' do
+                expect do
+                  subject.send(
```

Comment:
Rather than testing private methods (using subject.send - usually not great to do because it's an implementation detail) can you instead test the behavior using subject.call?

Reply:
If I were to test via using `subject.call`, I would have to change the [`SUPPORTED_ENCODING` constant](https://github.com/smithy-lang/smithy-ruby/blob/a30fa264f24147e058a28fd27e1b9829dc600fe4/hearth/lib/hearth/http/middleware/request_compression.rb#L10) (by using `.replace` since it is frozen). Since, changing the constants are highly discouraged - I opted for using `subject.send` instead. 

Let me know if there's another avenue that I can look at!

Reply:
Are you not able to use stub_const?

Reply:
`stub_const('Hearth::HTTP::Middleware::SUPPORTED_ENCODINGS', ['gzip', 'custom'])`

Reply:
Good to know for future! thanks